mkdocs/docs/HPC/setting_up_python_virtual_environments.md

```diff
@@ -0,0 +1,221 @@
+# Python Virtual Environments (venv's)
+
+## Introduction
+
+Python virtual environments (venv's) are a way to create isolated Python environments.
+You can install packages in a venv without affecting the system Python installation.
+Because a normal user cannot install packages globally, 
+venv's are a way to install packages without needing root access.
+This is useful when you need to use a package that is not available as a module on the HPC cluster.
+
+## Creating a venv
+
+A venv can be created with the following commands:
+
+```bash
+$ python -m venv myenv      # Create a new venv named myenv
+$ source myenv/bin/activate # Activate the venv
+```
+
+After activating the venv, you can install packages with `pip`:
+
+```bash
+$ pip install example_package1
+$ pip install example_package2
+```
+
+It is now possible to run Python scripts that use the installed packages in the venv. To deactivate the venv, run:
+
+```bash
+$ deactivate
+```
+
+!!! note
+    Always prefer to use modules if they are available. 
+    Modules are compiled and optimized for the HPC cluster, while packages installed with `pip` are not.
+    To check if a package is available as a module, you can use the following command:
+
+    ```bash
+    $ module av package_name
+    ```
+
+## Combining venv's with modules
+
+You can combine packages installed in a venv with modules. The following script uses 
+pytorch (which is available as a module) and Poutyne (which is not available as a module):
```

Comment:
`pytorch` -> `PyTorch` (the `torch` Python package)

Reply:
We may install `Poutyne` at some point, so at least say `which we assume is not centrally installed`, or use a fictional example Python packages which doesn't exist (and never will)

Reply:
Both fixed in [0f603ef](https://github.com/hpcugent/vsc_user_docs/pull/639/commits/0f603efaa8cda5010e587b7f49b43be61478fb0b)